docs/scripts/buildApi.ts

```diff
@@ -192,6 +192,71 @@ async function annotateComponentDefinition(context: {
   writeFileSync(typesFilename, typesSourceNew, { encoding: 'utf8' });
 }
 
+function trimComment(comment: string) {
+  let i = 0;
+  for (; i < comment.length; i++) {
+    if (comment[i] !== '*' && comment[i] !== ' ') {
+      break;
+    }
+  }
+  return comment.substr(i, comment.length - 1);
+}
+
+async function updateStylesDefinition(context: { api: ReactApi; component: { filename: string } }) {
+  const { api, component } = context;
+
+  const typesFilename = component.filename.replace(/\.js$/, '.d.ts');
+  const typesSource = readFileSync(typesFilename, { encoding: 'utf8' });
+  const typesAST = await babel.parseAsync(typesSource, {
+    configFile: false,
+    filename: typesFilename,
+    presets: [require.resolve('@babel/preset-typescript')],
+  });
+
+  if (typesAST === null) {
+    throw new Error('No AST returned from babel.');
+  }
+
+  if (api.styles.classes.length === 0) {
+    const componentName = path.basename(typesFilename).replace(/\.d\.ts$/, '');
+
+    (typesAST as any).program.body.forEach((node: any) => {
```

Comment:
Check out `annotateClassesDefinition` for how to traverse an AST.

Reply:
Refactored 👍